CMakeLists.txt

```diff
@@ -13,16 +12,14 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-# =======================================================================
 #
-
-cmake_minimum_required(VERSION 3.12.0)
+cmake_minimum_required(VERSION 3.13...3.17)
```

Comment:
Setting Min and Max versions to be used.

One of the features introduced in `version 3.13` is that target_sources doesn't have to be called in the same CMakeLists file that add_library is called.  

If you look at `lib/CMakeLists.txt`, there is no traces of call to target_sources and rather this is delegated to each sub-directories (e.g ae, base, fileio, listeners, nodes, etc...).

If you are interested, there is a great article on the advantages of target_sources and enhancements in `version 3.13`:
https://crascit.com/2016/01/31/enhanced-source-file-handling-with-target_sources/

Reply:
I don't think we necessarily need to update this periodically or at least any times soon to support newer features. 

The version range basically specifies that the current CMake code is written and tested for the given range of CMake versions and avoids having to setup the explicit policy settings. 

"max" compatibility version affects policy settings. So, if someone is using a version which is newer than <max-version> let's say 3.19 then they can set policies as far as the "max" version.

Please see this:
https://cmake.org/cmake/help/v3.12/command/cmake_minimum_required.html

Reply:
Ahh, sorry. See? I should have rtfm. :)

I foolishly assumed the max was a hard cutoff on which versions of CMake the project could be built with. Makes a lot more sense that it only determines policy settings instead. Duh...

Reply:
Lol, no worries.... I am no CMake expert myself at all and have to read the manuals from time to time to understand things :)